astroid/brain/brain_typing.py

```diff
@@ -132,6 +148,44 @@ def infer_typing_typevar_or_newtype(node, context_itton=None):
     return node.infer(context=context_itton)
 
 
+def infer_typing_newtype(
+    node: nodes.Call, context_itton: typing.Optional[context.InferenceContext] = None
+) -> typing.Iterator[nodes.ClassDef]:
+    """Infer a typing.NewType(...) call"""
+    try:
+        func = next(node.func.infer(context=context_itton))
+    except (InferenceError, StopIteration) as exc:
+        raise UseInferenceDefault from exc
+
+    if func.qname() != "typing.NewType":
+        raise UseInferenceDefault
+    if len(node.args) != 2:
+        raise UseInferenceDefault
+
+    derived, base = node.args
+    derived_name = derived.as_string().strip("'")
+    base_name = base.as_string().strip("'")
+
+    new_node: ClassDef = extract_node(
+        TYPING_NEWTYPE_TEMPLATE.format(derived=derived_name, base=base_name)
+    )
+    new_node.parent = node.parent
+
+    # Base type arg is a normal reference, so no need to do special lookups
+    if not isinstance(base, nodes.Const):
+        new_node.bases = [base]
+
+    # If the base type is given as a string (e.g. for a forward reference),
+    # make a naive attempt to find the corresponding node.
+    # Note that this will not work with imported types.
```

Comment:
This is the one remaining edge case I ran into. It comes up with code such as:

```python
from decimal import Decimal
A = NewType("A", "Decimal")
```

or

```python
import decimal
A = NewType("A", "decimal.Decimal")
```

The `node.frame().lookup()` won't find anything that can be used directly as a base. In the first case, I was able to hack something up by recursively resolving the `ImportFrom`'s module and doing a `lookup()` on that. In the latter, I wasn't able to figure out a clean solution.

Is there any comprehensive method to resolve these types of string references? If not, what would be the recommended course of action here? Worst case, I can also add a test and mark it as xfail to improve documentation of the limitation.

Reply:
I'm not that experienced with `astroid` brains, so I'm not sure honestly.

I think this is already seems like a good starting approach.

Reply:
Having thought about this a bit more:
Would splitting on `"."`, looking op `decimal` and then looking up `Decimal` work? Perhaps you have already tried this. If so, I'll see if I can come up with something else.

Reply:
Yeah I was playing with doing essentially that. There weren't any major problems, I just wasn't sure if there was a cleaner approach. I'll revive that code.

Reply:
I took a shot at implementing this, let me know what you think. Definitely a bit hairy, but it seems to work for most cases.